src/test/java/org/xembly/XemblerTest.java

```diff
@@ -177,4 +185,67 @@ public void replacesRootNode() throws Exception {
         );
     }
 
+    /**
+     * Test that concurrent invocations on shared DOM.
+     * doesn't ruin executing thread
+     * @todo #34
+     *  I assumed that Xembler should be a thread-safe and
+     *  this test must pass, but this assumption might be wrong and
+     *  this test case is not an issue
+     * @throws Exception If some problem inside
+     */
+    @Test
+    @Ignore
+    public void concurrentInvocationWithNoExceptions() throws Exception {
+        final ExecutorService service = Executors.newFixedThreadPool(
+                Runtime.getRuntime().availableProcessors()
+        );
+        final Document dom = DocumentBuilderFactory.newInstance()
+                .newDocumentBuilder().newDocument();
+        final Xembler xembler = new Xembler(
+            new Directives(
+                "ADD 'root';ADDIF 'blow';"
+            )
+        );
+        final int capacity = 10000;
+        final Collection<Callable<Document>> tasks =
+            new ArrayList<Callable<Document>>(
+                    capacity
+            );
+        for (int idx = 0; idx < capacity; ++idx) {
+            final Callable<Document> callable = XemblerTest.buildCallable(
+                    xembler, dom
+            );
+            tasks.add(callable);
+        }
+        final List<Future<Document>> futures = service.invokeAll(tasks);
+        for (final Future<Document> future : futures) {
+            final Document document = future.get();
+            MatcherAssert.assertThat(
+                XhtmlMatchers.xhtml(document),
+                XhtmlMatchers.hasXPaths(
+                        "/root[count(blow) = 1]"
+                )
+            );
+        }
+        service.shutdown();
+    }
+
+    /**
+     * Builds up callable object.
+     * @param xembler Xembler's instance
+     * @param document DOM object
+     * @return Callable object
+     */
+    private static Callable<Document> buildCallable(
```

Comment:
I would name it `callable`, see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#method-names